compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/platform/Accessibility.uikit.kt

```diff
@@ -186,170 +245,291 @@ private class AccessibilityElement(
+            parent?.scrollByIfPossible(0f, unclippedRect.bottom - scrollableAncestorRect.bottom)
+            return
+        } else if (unclippedRect.left < scrollableAncestorRect.left) {
+            // The element is to the left of the screen, scroll left
+            parent?.scrollByIfPossible(unclippedRect.left - scrollableAncestorRect.left, 0f)
+            return
+        } else if (unclippedRect.right > scrollableAncestorRect.right) {
+            // The element is to the right of the screen, scroll right
+            parent?.scrollByIfPossible(unclippedRect.right - scrollableAncestorRect.right, 0f)
+            return
+        }
+    }
 
-        fun onMeaningfulSemanticAdded() {
-            hasAnyMeaningfulSemantics = true
+    private fun scrollByIfPossible(dx: Float, dy: Float) {
+        if (!isAlive) {
+            return
         }
 
-        val accessibilityLabelStrings = mutableListOf<String>()
-        val accessibilityValueStrings = mutableListOf<String>()
-        var accessibilityTraits = UIAccessibilityTraitNone
+        // if has scrollBy action, invoke it, otherwise try to scroll the parent
+        val action = semanticsNode.config.getOrNull(SemanticsActions.ScrollBy)?.action
 
-        fun addTrait(trait: UIAccessibilityTraits) {
-            accessibilityTraits = accessibilityTraits or trait
+        if (action != null) {
+            action(dx, dy)
+        } else {
+            parent?.scrollByIfPossible(dx, dy)
         }
+    }
+
+    private fun scrollIfPossible(direction: UIAccessibilityScrollDirection, config: SemanticsConfiguration): Boolean {
+        val (width, height) = semanticsNode.size
 
-        fun <T> getValue(key: SemanticsPropertyKey<T>): T = semanticsNode.config[key]
+        // TODO: reverse engineer proper dimension scale
+        val dimensionScale = 0.5f
 
-        // Iterate through all semantic properties and map them to values that are expected by iOS Accessibility services for the node with given semantics
-        semanticsNode.config.forEach { pair ->
-            when (val key = pair.key) {
-                // == Properties ==
+        // TODO: post notification about the scroll
+        when (direction) {
+            UIAccessibilityScrollDirectionUp -> {
+                var result = config.getOrNull(SemanticsActions.PageUp)?.action?.invoke()
 
-                SemanticsProperties.InvisibleToUser -> {
-                    // Return immediately. Function won't reach a point where [isAccessibilityElement] is set to true
-                    return
+                if (result != null) {
+                    return result
                 }
 
-                SemanticsProperties.LiveRegion -> {
-                    onMeaningfulSemanticAdded()
-                    addTrait(UIAccessibilityTraitUpdatesFrequently)
+                result = config.getOrNull(SemanticsActions.ScrollBy)?.action?.invoke(
+                    0F,
+                    -height.toFloat() * dimensionScale
```

Comment:
can we add argument names here? Or it is Function type without named arguments?

Reply:
Yes, it's `(x: Float, y: Float) -> Boolean`, IDE can recognise the labels, but explicit named arguments are not allowed for function types indeed.